Replace the OSM-iMX93 LPDDR4X timing table with the DDR Tool generated
table for the 512MB part.

Generated with DDR Tool v4.0.0_8.8-1eade933a (vendor drop OSM-iMX93_DDR_512M).
This supersedes the earlier stopgap that only capped REG_DDR_CS0_BNDS on the
2GB table: the table below carries the correct density (0x4e300000 = 0x80009f),
address map (0x4e300080 = 0x80000312), refresh timings and PHY training for a
4Gb die.

Note the frequency setpoints change from { 3200, 1600, 625 } to
{ 3733, 1866, 625 }.

Upstream-Status: Inappropriate [board specific]
--- a/board/adlink/osm-imx93/lpddr4x_timing.c
+++ b/board/adlink/osm-imx93/lpddr4x_timing.c
@@ -1,10 +1,11 @@
 /*
- * Copyright 2025 NXP
+ * Copyright 2026 NXP
  *
  * SPDX-License-Identifier: BSD-3-Clause
  *
- * Code generated with DDR Tool v3.5.0_8.4-1ddf053d.
- * DDR PHY FW2022.01
+ * Code generated with DDR Tool v4.0.0_8.8-1eade933a.
+ * DDR PHY FW2022.01 
+ * Part number: NXP LPDDR4X EVK and FRDM 11x11 board's default DDR part
  */
 
 #include <linux/kernel.h>
@@ -13,9 +14,9 @@
 /* Initialize DDRC registers */
 static struct dram_cfg_param ddr_ddrc_cfg[] = {
     {0x4e300110, 0x44100001},
-    {0x4e300000, 0x8000ff},
+    {0x4e300000, 0x80009f},
     {0x4e300008, 0x0},
-    {0x4e300080, 0x80000512},
+    {0x4e300080, 0x80000312},
     {0x4e300084, 0x0},
     {0x4e300114, 0x1002},
     {0x4e300260, 0x80},
@@ -37,81 +38,81 @@
 static struct dram_fsp_cfg ddr_dram_fsp_cfg[] = {
     {
         {
-            {0x4e300100, 0x1465311B},
-            {0x4e300104, 0xF8CC0017},
-            {0x4e300108, 0xD4D68E00},
-            {0x4e30010C, 0x0071E100},
-            {0x4e300124, 0x18660000},
-            {0x4e300160, 0x00009102},
-            {0x4e30016C, 0x33F00000},
+            {0x4e300100, 0x2494321B},
+            {0x4e300104, 0xDAEE001B},
+            {0x4e300108, 0x2F2E8233},
+            {0x4e30010C, 0x0005E18B},
+            {0x4e300124, 0x1C030000},
+            {0x4e300160, 0x0000B101},
+            {0x4e30016C, 0x35F00000},
             {0x4e300170, 0x8B0B0608},
-            {0x4e300250, 0x00000022},
-            {0x4e300254, 0x01280128},
+            {0x4e300250, 0x00000028},
+            {0x4e300254, 0x00A000A0},
             {0x4e300258, 0x00000008},
             {0x4e30025C, 0x00000400},
-            {0x4e300300, 0x1D441D0D},
-            {0x4e300304, 0x01281E10},
-            {0x4e300308, 0x0C3C0C34},
+            {0x4e300300, 0x224F2213},
+            {0x4e300304, 0x00A02213},
+            {0x4e300308, 0x0A3C0E03},
         },
         {
-            {0x01, 0xD4},
-            {0x02, 0x2D},
-            {0x03, 0x33},
+            {0x01, 0xE4},
+            {0x02, 0x36},
+            {0x03, 0x32},
             {0x0b, 0x46},
-            {0x0c, 0x48},
-            {0x0e, 0x48},
+            {0x0c, 0x11},
+            {0x0e, 0x11},
             {0x16, 0x04},
         },
         0,
     },
     {
         {
-            {0x4e300100, 0x02121100},
-            {0x4e300104, 0xF866000C},
-            {0x4e300108, 0xF2FA8088},
-            {0x4e30010C, 0x006101A0},
-            {0x4e300124, 0x0C340000},
-            {0x4e300160, 0x00009101},
-            {0x4e30016C, 0x30700000},
+            {0x4e300100, 0x124A2100},
+            {0x4e300104, 0xDA77000E},
+            {0x4e300108, 0x181604AA},
+            {0x4e30010C, 0x005101E6},
+            {0x4e300124, 0x0E030000},
+            {0x4e300160, 0x0000B100},
+            {0x4e30016C, 0x30900000},
             {0x4e300170, 0x8A0A0508},
-            {0x4e300250, 0x00000011},
-            {0x4e300254, 0x00900090},
+            {0x4e300250, 0x00000014},
+            {0x4e300254, 0x004C004C},
             {0x4e300258, 0x00000008},
             {0x4e30025C, 0x00000400},
         },
         {
-            {0x01, 0xA4},
-            {0x02, 0x52},
-            {0x03, 0x33},
+            {0x01, 0xB4},
+            {0x02, 0x1B},
+            {0x03, 0x32},
             {0x0b, 0x46},
-            {0x0c, 0x48},
-            {0x0e, 0x48},
+            {0x0c, 0x11},
+            {0x0e, 0x11},
             {0x16, 0x04},
         },
         0,
     },
     {
         {
-            {0x4e300100, 0x00061000},
-            {0x4e300104, 0xF855000A},
-            {0x4e300108, 0x6E62FA48},
+            {0x4e300100, 0x00031000},
+            {0x4e300104, 0xDA55000A},
+            {0x4e300108, 0x6E621A48},
             {0x4e30010C, 0x0031010D},
-            {0x4e300124, 0x04C50000},
-            {0x4e300160, 0x00009100},
+            {0x4e300124, 0x04830000},
+            {0x4e300160, 0x0000B100},
             {0x4e30016C, 0x30000000},
             {0x4e300170, 0x89090408},
             {0x4e300250, 0x00000007},
-            {0x4e300254, 0x00340034},
+            {0x4e300254, 0x00150015},
             {0x4e300258, 0x00000008},
             {0x4e30025C, 0x00000400},
         },
         {
             {0x01, 0x94},
             {0x02, 0x9},
-            {0x03, 0x33},
+            {0x03, 0x32},
             {0x0b, 0x46},
-            {0x0c, 0x48},
-            {0x0e, 0x48},
+            {0x0c, 0x11},
+            {0x0e, 0x11},
             {0x16, 0x04},
         },
         1,
@@ -137,18 +138,18 @@
     {0x110a5, 0x5},
     {0x110a6, 0x6},
     {0x110a7, 0x7},
-    {0x1005f, 0x1ff},
-    {0x1015f, 0x1ff},
-    {0x1105f, 0x1ff},
-    {0x1115f, 0x1ff},
-    {0x11005f, 0x1ff},
-    {0x11015f, 0x1ff},
-    {0x11105f, 0x1ff},
-    {0x11115f, 0x1ff},
-    {0x21005f, 0x1ff},
-    {0x21015f, 0x1ff},
-    {0x21105f, 0x1ff},
-    {0x21115f, 0x1ff},
+    {0x1005f, 0x5ff},
+    {0x1015f, 0x5ff},
+    {0x1105f, 0x5ff},
+    {0x1115f, 0x5ff},
+    {0x11005f, 0x5ff},
+    {0x11015f, 0x5ff},
+    {0x11105f, 0x5ff},
+    {0x11115f, 0x5ff},
+    {0x21005f, 0x5ff},
+    {0x21015f, 0x5ff},
+    {0x21105f, 0x5ff},
+    {0x21115f, 0x5ff},
     {0x55, 0x1ff},
     {0x1055, 0x1ff},
     {0x2055, 0x1ff},
@@ -156,7 +157,7 @@
     {0x1200c5, 0xb},
     {0x2200c5, 0x7},
     {0x2002e, 0x2},
-    {0x12002e, 0x1},
+    {0x12002e, 0x2},
     {0x22002e, 0x2},
     {0x90204, 0x0},
     {0x190204, 0x0},
@@ -188,27 +189,27 @@
     {0x21014d, 0x600},
     {0x21104d, 0x600},
     {0x21114d, 0x600},
-    {0x10049, 0xe3f},
-    {0x10149, 0xe3f},
-    {0x11049, 0xe3f},
-    {0x11149, 0xe3f},
-    {0x110049, 0xe3f},
-    {0x110149, 0xe3f},
-    {0x111049, 0xe3f},
-    {0x111149, 0xe3f},
-    {0x210049, 0xe3f},
-    {0x210149, 0xe3f},
-    {0x211049, 0xe3f},
-    {0x211149, 0xe3f},
-    {0x43, 0x7f},
-    {0x1043, 0x7f},
-    {0x2043, 0x7f},
+    {0x10049, 0xe00},
+    {0x10149, 0xe00},
+    {0x11049, 0xe00},
+    {0x11149, 0xe00},
+    {0x110049, 0xe00},
+    {0x110149, 0xe00},
+    {0x111049, 0xe00},
+    {0x111149, 0xe00},
+    {0x210049, 0xe00},
+    {0x210149, 0xe00},
+    {0x211049, 0xe00},
+    {0x211149, 0xe00},
+    {0x43, 0x60},
+    {0x1043, 0x60},
+    {0x2043, 0x60},
     {0x20018, 0x1},
     {0x20075, 0x4},
-    {0x20050, 0x11},
+    {0x20050, 0x0},
     {0x2009b, 0x2},
-    {0x20008, 0x320},
-    {0x120008, 0x190},
+    {0x20008, 0x3a5},
+    {0x120008, 0x1d3},
     {0x220008, 0x9c},
     {0x20088, 0x9},
     {0x200b2, 0x10c},
@@ -1237,7 +1238,7 @@
 /* P0 message block parameter for training firmware */
 static struct dram_cfg_param ddr_fsp0_cfg[] = {
     {0xd0000, 0x0},
-    {0x54003, 0xc80},
+    {0x54003, 0xe94},
     {0x54004, 0x4},
     {0x54006, 0x15},
     {0x54008, 0x131f},
@@ -1246,27 +1247,27 @@
     {0x5400d, 0x100},
     {0x5400f, 0x100},
     {0x54012, 0x110},
-    {0x54019, 0x2dd4},
-    {0x5401a, 0x33},
-    {0x5401b, 0x4846},
-    {0x5401c, 0x4808},
+    {0x54019, 0x36e4},
+    {0x5401a, 0x32},
+    {0x5401b, 0x1146},
+    {0x5401c, 0x1108},
     {0x5401e, 0x4},
-    {0x5401f, 0x2dd4},
-    {0x54020, 0x33},
-    {0x54021, 0x4846},
-    {0x54022, 0x4808},
+    {0x5401f, 0x36e4},
+    {0x54020, 0x32},
+    {0x54021, 0x1146},
+    {0x54022, 0x1108},
     {0x54024, 0x4},
-    {0x54032, 0xd400},
-    {0x54033, 0x332d},
+    {0x54032, 0xe400},
+    {0x54033, 0x3236},
     {0x54034, 0x4600},
-    {0x54035, 0x848},
-    {0x54036, 0x48},
+    {0x54035, 0x811},
+    {0x54036, 0x11},
     {0x54037, 0x400},
-    {0x54038, 0xd400},
-    {0x54039, 0x332d},
+    {0x54038, 0xe400},
+    {0x54039, 0x3236},
     {0x5403a, 0x4600},
-    {0x5403b, 0x848},
-    {0x5403c, 0x48},
+    {0x5403b, 0x811},
+    {0x5403c, 0x11},
     {0x5403d, 0x400},
     {0xd0000, 0x1}
 };
@@ -1275,7 +1276,7 @@
 static struct dram_cfg_param ddr_fsp1_cfg[] = {
     {0xd0000, 0x0},
     {0x54002, 0x1},
-    {0x54003, 0x640},
+    {0x54003, 0x74a},
     {0x54004, 0x4},
     {0x54006, 0x15},
     {0x54008, 0x121f},
@@ -1284,27 +1285,27 @@
     {0x5400d, 0x100},
     {0x5400f, 0x100},
     {0x54012, 0x110},
-    {0x54019, 0x52a4},
-    {0x5401a, 0x33},
-    {0x5401b, 0x4846},
-    {0x5401c, 0x4808},
+    {0x54019, 0x1bb4},
+    {0x5401a, 0x32},
+    {0x5401b, 0x1146},
+    {0x5401c, 0x1108},
     {0x5401e, 0x4},
-    {0x5401f, 0x52a4},
-    {0x54020, 0x33},
-    {0x54021, 0x4846},
-    {0x54022, 0x4808},
+    {0x5401f, 0x1bb4},
+    {0x54020, 0x32},
+    {0x54021, 0x1146},
+    {0x54022, 0x1108},
     {0x54024, 0x4},
-    {0x54032, 0xa400},
-    {0x54033, 0x3352},
+    {0x54032, 0xb400},
+    {0x54033, 0x321b},
     {0x54034, 0x4600},
-    {0x54035, 0x848},
-    {0x54036, 0x48},
+    {0x54035, 0x811},
+    {0x54036, 0x11},
     {0x54037, 0x400},
-    {0x54038, 0xa400},
-    {0x54039, 0x3352},
+    {0x54038, 0xb400},
+    {0x54039, 0x321b},
     {0x5403a, 0x4600},
-    {0x5403b, 0x848},
-    {0x5403c, 0x48},
+    {0x5403b, 0x811},
+    {0x5403c, 0x11},
     {0x5403d, 0x400},
     {0xd0000, 0x1}
 };
@@ -1323,26 +1324,26 @@
     {0x5400f, 0x100},
     {0x54012, 0x110},
     {0x54019, 0x994},
-    {0x5401a, 0x33},
-    {0x5401b, 0x4846},
-    {0x5401c, 0x4800},
+    {0x5401a, 0x32},
+    {0x5401b, 0x1146},
+    {0x5401c, 0x1100},
     {0x5401e, 0x4},
     {0x5401f, 0x994},
-    {0x54020, 0x33},
-    {0x54021, 0x4846},
-    {0x54022, 0x4800},
+    {0x54020, 0x32},
+    {0x54021, 0x1146},
+    {0x54022, 0x1100},
     {0x54024, 0x4},
     {0x54032, 0x9400},
-    {0x54033, 0x3309},
+    {0x54033, 0x3209},
     {0x54034, 0x4600},
-    {0x54035, 0x48},
-    {0x54036, 0x48},
+    {0x54035, 0x11},
+    {0x54036, 0x11},
     {0x54037, 0x400},
     {0x54038, 0x9400},
-    {0x54039, 0x3309},
+    {0x54039, 0x3209},
     {0x5403a, 0x4600},
-    {0x5403b, 0x48},
-    {0x5403c, 0x48},
+    {0x5403b, 0x11},
+    {0x5403c, 0x11},
     {0x5403d, 0x400},
     {0xd0000, 0x1}
 };
@@ -1351,7 +1352,7 @@
 /* P0 2D message block parameter for training firmware */
 static struct dram_cfg_param ddr_fsp0_2d_cfg[] = {
     {0xd0000, 0x0},
-    {0x54003, 0xc80},
+    {0x54003, 0xe94},
     {0x54004, 0x4},
     {0x54006, 0x15},
     {0x54008, 0x61},
@@ -1361,27 +1362,27 @@
     {0x5400f, 0x100},
     {0x54010, 0x2080},
     {0x54012, 0x110},
-    {0x54019, 0x2dd4},
-    {0x5401a, 0x33},
-    {0x5401b, 0x4846},
-    {0x5401c, 0x4808},
+    {0x54019, 0x36e4},
+    {0x5401a, 0x32},
+    {0x5401b, 0x1146},
+    {0x5401c, 0x1108},
     {0x5401e, 0x4},
-    {0x5401f, 0x2dd4},
-    {0x54020, 0x33},
-    {0x54021, 0x4846},
-    {0x54022, 0x4808},
+    {0x5401f, 0x36e4},
+    {0x54020, 0x32},
+    {0x54021, 0x1146},
+    {0x54022, 0x1108},
     {0x54024, 0x4},
-    {0x54032, 0xd400},
-    {0x54033, 0x332d},
+    {0x54032, 0xe400},
+    {0x54033, 0x3236},
     {0x54034, 0x4600},
-    {0x54035, 0x848},
-    {0x54036, 0x48},
+    {0x54035, 0x811},
+    {0x54036, 0x11},
     {0x54037, 0x400},
-    {0x54038, 0xd400},
-    {0x54039, 0x332d},
+    {0x54038, 0xe400},
+    {0x54039, 0x3236},
     {0x5403a, 0x4600},
-    {0x5403b, 0x848},
-    {0x5403c, 0x48},
+    {0x5403b, 0x811},
+    {0x5403c, 0x11},
     {0x5403d, 0x400},
     {0xd0000, 0x1}
 };
@@ -1871,13 +1872,13 @@
     {0x400d7, 0x20b},
     {0x2003a, 0x2},
     {0x200be, 0x3},
-    {0x2000b, 0x384},
-    {0x2000c, 0xc8},
-    {0x2000d, 0x7d0},
+    {0x2000b, 0x41a},
+    {0x2000c, 0xe9},
+    {0x2000d, 0x91c},
     {0x2000e, 0x2c},
-    {0x12000b, 0x1c2},
-    {0x12000c, 0x64},
-    {0x12000d, 0x3e8},
+    {0x12000b, 0x20d},
+    {0x12000c, 0x74},
+    {0x12000d, 0x48e},
     {0x12000e, 0x2c},
     {0x22000b, 0xb0},
     {0x22000c, 0x27},
@@ -1951,15 +1952,15 @@
 
 static struct dram_fsp_msg ddr_dram_fsp_msg[] = {
     {
-        /* P0 3200mts 1D */
-        .drate = 3200,
+        /* P0 3733mts 1D */
+        .drate = 3733,
         .fw_type = FW_1D_IMAGE,
         .fsp_cfg = ddr_fsp0_cfg,
         .fsp_cfg_num = ARRAY_SIZE(ddr_fsp0_cfg),
     },
     {
-        /* P1 1600mts 1D */
-        .drate = 1600,
+        /* P1 1866mts 1D */
+        .drate = 1866,
         .fw_type = FW_1D_IMAGE,
         .fsp_cfg = ddr_fsp1_cfg,
         .fsp_cfg_num = ARRAY_SIZE(ddr_fsp1_cfg),
@@ -1972,8 +1973,8 @@
         .fsp_cfg_num = ARRAY_SIZE(ddr_fsp2_cfg),
     },
     {
-        /* P0 3200mts 2D */
-        .drate = 3200,
+        /* P0 3733mts 2D */
+        .drate = 3733,
         .fw_type = FW_2D_IMAGE,
         .fsp_cfg = ddr_fsp0_2d_cfg,
         .fsp_cfg_num = ARRAY_SIZE(ddr_fsp0_2d_cfg),
@@ -1992,7 +1993,7 @@
     .ddrphy_trained_csr_num = ARRAY_SIZE(ddr_ddrphy_trained_csr),
     .ddrphy_pie = ddr_phy_pie,
     .ddrphy_pie_num = ARRAY_SIZE(ddr_phy_pie),
-    .fsp_table = { 3200, 1600, 625, },
+    .fsp_table = { 3733, 1866, 625, },
     .fsp_cfg = ddr_dram_fsp_cfg,
     .fsp_cfg_num = ARRAY_SIZE(ddr_dram_fsp_cfg),
 };
